latest_major_version.go

```diff
@@ -0,0 +1,112 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"golang.org/x/mod/modfile"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+)
+
+func getLatestMajorVersion(repo string) (string, error) {
+	// Get latest major version available for repo from github.
+	const apiURL = "https://api.github.com/repos/%s/tags"
+	url := fmt.Sprintf(apiURL, repo)
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("failed to fetch tags: %s", resp.Status)
+	}
+
+	var tags []struct {
+		Name string `json:"name"`
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(&tags); err != nil {
+		return "", err
+	}
+
+	latest := ""
+	// match on version strings e.x. v9.3.0
+	majorVersionRegex := regexp.MustCompile(`^v(\d+)`)
+	alphaBetaRegex := regexp.MustCompile(`alpha|beta`)
+
+	// Check latest major version seen
+	for _, tag := range tags {
+		if majorVersionRegex.MatchString(tag.Name) && !alphaBetaRegex.MatchString(tag.Name) {
+			version := majorVersionRegex.FindString(tag.Name)
+			if latest == "" || version > latest {
```

Comment:
Thoughts on instead of comparing `string` values we convert to `int` and compare on that?

something like so (ignore my bad `go` code:

```go
matches := majorVersionRegex.FindStringSubmatch(tag.Name)
if (len(matches) > 1)
{
   version, err := strconv.Atoi(matches[1])
   if err != nil {
     continue;
   }
   if version > latest {
     latestVersion = version
   }
}
```

I think one issue with string comparison is that you'll get things like `v11` is less than `v2`? (but this may have already been accounted for)

Maybe instead we could also look into a semantic versioning package? Like: https://github.com/Masterminds/semver?

Reply:
I think the current one definitely works though as we were getting things like that `redis` detection for the new v9

And double majors are probably not common, so this can be done at a later point

Reply:
Done!